src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/LayoutPath.java

```diff
@@ -72,6 +73,8 @@
                     MethodType.methodType(long.class, MemorySegment.class, long.class, long.class, long.class));
             MH_ADD_SCALED_OFFSET = lookup.findStatic(LayoutPath.class, "addScaledOffset",
                     MethodType.methodType(long.class, long.class, long.class, long.class));
+            MH_SLICE = lookup.findStatic(LayoutPath.class, "slice",
```

Comment:
Why not just pointing to MS::asSlice?

Reply:
Yeah, I'll do that. I did it like this at first just in case we wanted to change the slice API in the future, but looking now it seems kinda YAGNI